scripts/gdx/gdxAirdropCalculation.ts

```diff
@@ -198,3 +210,224 @@ export const airdrop = (ethers: typeof Ethers, ethSnapshotBlock) => {
+    await populateListOfAddressesAndBalances(exchangeHelper, exchangeHelperTokenPurchasedFilter, exchangeHelperTokenSoldFilter);
+
+    delete newAddresses[exchangeHelper.address];
+
+    // console.log({ newAddresses, newIsContracts });
+    return { newAddresses, newIsContracts };
+  };
+
+  const buildMerkleTree = () => {
+    const { addressesCombined, isContracts } = JSON.parse(
+      fs.readFileSync("airdrop/buyBalancesCombined.json").toString()
+    );
+    
+    let toTree: Array<[string, BigNumber]> = Object.entries(addressesCombined).map(
+      ([addr, gdx]) => {
+        return [addr, gdx as BigNumber];
+      }
+    );
+    
+    // console.log(`Before sorting`);
+    // toTree.forEach((a,_) => { console.log(`${a[0].toString()}:${ethers.BigNumber.from(a[1]).toString()}\n`)}); 
+    
+    toTree.sort((a, b) => BigNumber.from(a[1]).sub(b[1]) < ZERO ? 0 : -1 );
+
+    // console.log(`After sorting`);
+    // toTree.forEach((a,_) => { console.log(`${a[0].toString()}:${ethers.BigNumber.from(a[1]).toString()}\n`)}); 
+    
+    // toTree.forEach((a,_) => console.log(`${a[0].toString()},${BigNumber.from(a[1]).toString()}`));
+    
+    let totalGDX = ZERO;
+    toTree.forEach((a,_) => totalGDX = totalGDX.add(a[1]))
+    console.log({
+      isContracts,
+      toTree: toTree.forEach((a,_) => console.log({address: a[0].toString(), balance: BigNumber.from(a[1]).toString()})),
+      numberOfAccounts: toTree.length,
+      TotalGDX: totalGDX.toString()
+    });
+
+    // Print statistics
+    const sorted = toTree.map(_ => _[1]);
+    console.log("GDX Distribution\n");
+    [0.001, 0.01, 0.1, 0.5].forEach(q =>
+      console.log({
+        precentile: q * 100 + "%",
+        gdx: quantileBN(sorted, q)
+      })
+    );
+
+    const treeData = {};
+    const elements = toTree.map(e => {
+      const hash = ethers.utils.keccak256(
+        ethers.utils.defaultAbiCoder.encode(
+          ["address", "uint256"],
+          [e[0], e[1]]
+        )
+      );
+      treeData[e[0]] = {
+        gdx: e[1],
+        hash
+      };
+      return Buffer.from(hash.slice(2), "hex");
+    });
+
+    console.log(elements);
+    const merkleTree = new MerkleTree(elements, false);
+    // get the merkle root
+    // returns 32 byte buffer
+    const merkleRoot = merkleTree.getRoot().toString("hex");
+    // generate merkle proof
+    // returns array of 32 byte buffers
+    const proof = merkleTree.getProof(elements[0]).map(_ => _.toString("hex"));
+    console.log({ merkleRoot, proof, sampleProofFor: toTree[0] });
+    fs.writeFileSync(
+      "airdrop/gdxairdropNew.json",
+      JSON.stringify({ treeData, merkleRoot })
+    );
+  };
+
+  const addCalculationsToPreviousData = async () => {
+    // get previous airdrop and turn it into BigNumbers
+    const { addresses, isContracts } = JSON.parse(
+      fs.readFileSync("airdrop/buyBalances.json").toString()
```

Comment:
is that the file the issue specify to use?

Reply:
gdxairdrop.json is, changing to it